addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -24,6 +25,38 @@ odoo.define('point_of_sale.ProductsWidgetControlPanel', function(require) {
                 this.trigger('try-add-product', { searchWordInput: this.searchWordInput });
             }
         }
+        async loadProductFromDB() {
+            if(!this.searchWordInput.el.value)
+                return;
+
+            try {
+                let ProductIds = await this.rpc({
+                    model: 'product.product',
+                    method: 'search',
+                    args: [[['name', 'ilike', this.searchWordInput.el.value + "%"]]],
```

Comment:
don't forget to add the common domain in model.js (like pos_available = True ...)